presto-main/src/main/java/com/facebook/presto/catalogserver/CatalogServer.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.catalogserver;
+
+import com.facebook.drift.annotations.ThriftMethod;
+import com.facebook.drift.annotations.ThriftService;
+import com.facebook.presto.SessionRepresentation;
+import com.facebook.presto.common.QualifiedObjectName;
+import com.facebook.presto.metadata.Catalog;
+import com.facebook.presto.metadata.CatalogManager;
+import com.facebook.presto.metadata.SessionPropertyManager;
+import com.facebook.presto.spi.ConnectorId;
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.ConnectorTableHandle;
+import com.facebook.presto.spi.connector.Connector;
+import com.facebook.presto.spi.connector.ConnectorMetadata;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import javax.inject.Inject;
+
+import java.util.Optional;
+
+import static com.facebook.presto.metadata.MetadataUtil.toSchemaTableName;
+import static com.facebook.presto.spi.transaction.IsolationLevel.SERIALIZABLE;
+import static java.util.Objects.requireNonNull;
+
+@ThriftService(value = "presto-catalog-server", idlName = "PrestoCatalogServer")
+public class CatalogServer
+{
+    private final SessionPropertyManager sessionPropertyManager;
+    private final CatalogManager catalogManager;
+
+    @Inject
+    public CatalogServer(SessionPropertyManager sessionPropertyManager, CatalogManager catalogManager)
+    {
+        this.sessionPropertyManager = requireNonNull(sessionPropertyManager, "sessionPropertyManager is null");
+        this.catalogManager = requireNonNull(catalogManager, "catalogMetadataManager is null");
+    }
+
+    @ThriftMethod
+    public boolean catalogExists(String catalogName)
+    {
+        return catalogManager.getCatalog(catalogName).isPresent();
+    }
+
+    @ThriftMethod
+    public boolean schemaExists(SessionRepresentation session, String catalogName, String schemaName)
```

Comment:
The session can contain lots of values. Passing the entire session for each CatalogServer call may be too heavy.

Reply:
Not sure if we can avoid session. It is required by the lowest level [ConnectorMetadata](https://github.com/prestodb/presto/blob/master/presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorMetadata.java).